src/iuse.cpp

```diff
@@ -2583,6 +2583,30 @@ int iuse::hammer(player *p, item *it, bool, const tripoint& )
     return 0;
 }
 
+int iuse::deploy( player *p, item *it, bool, const tripoint &pos )
+{
+    tripoint dir = pos;
+    if( pos == p->pos() ) {
+        if( !choose_adjacent( _( "Deploy where?" ), dir ) ) {
+            return 0;
+        }
+    }
+
+    if( dir.x == p->posx() && dir.y == p->posy() ) {
+        p->add_msg_if_player( m_info,
+                              _( "You attempt to become one with the furniture.  It doesn't work." ) );
+        return 0;
+    }
+
+    if( g->m.move_cost( dir ) != 2 ) {
```

Comment:
You should also check that there is no furniture there at the current time, as it would be lost when you call `furn_set`.

But this whole code would be much better as a dedicated `iuse_actor` type. This would also allow to move the `deploy_furn` member into that actor. Most tools won't need that member, so why should they have it?